### PR TITLE
Fix memory leak

### LINKED ISSUE #5810: Memory gradually increasing until crash while training simple MLP on CPU
I am running this [tutorial](http://deeplearning.net/tutorial/mlp.html) that is training a simple MLP on MNIST. I run on CPU. While monitoring my computer activity in the training loop, I can see that the memory usage is steadily increasing, until it reaches 100% and everything crashes (need to reboot).

I am on Ubuntu 16.04 and using the bleeding edge version of Theano.
There has been a similar [post](https://github.com/Theano/Theano/issues/4101) but the problem seemed to arise exclusively on GPU.

Is anyone experiencing the same issue on CPU?

### PR DESCRIPTION
**Pull request by lamblin**
Fix #5810, a memory leak when `blas.ldflags == ''`.

### CODE REVIEW

**Comment by notoraptor** [comment]
I pushed a commit to update sphinx version installed on TRAVIV for Python 3.3 . It seems sphinx version 1.5.1 is not available for this Python version:

```
(pyenv) ~/mila/dev/git/.parallel/Theano$ conda search sphinx 
Fetching package metadata .........
sphinx                       1.1.3                    py26_1  defaults        
                             1.1.3                    py27_1  defaults        
                             1.1.3                    py27_2  defaults        
                             1.1.3                    py33_2  defaults        
                             1.1.3                    py26_2  defaults        
                             1.1.3                    py27_3  defaults        
                             1.1.3                    py26_3  defaults        
                             1.1.3                    py33_3  defaults        
                             1.1.3                    py27_4  defaults        
                             1.1.3                    py33_4  defaults        
                             1.1.3                    py26_4  defaults        
                             1.2                      py33_0  defaults        
                             1.2                      py27_0  defaults        
                             1.2                      py26_0  defaults        
                             1.2.1                    py33_0  defaults        
                             1.2.1                    py27_0  defaults        
                             1.2.1                    py26_0  defaults        
                             1.2.2                    py34_0  defaults        
                             1.2.2                    py26_0  defaults        
                             1.2.2                    py27_0  defaults        
                             1.2.2                    py33_0  defaults        
                             1.2.2                    py26_1  defaults        
                             1.2.2                    py27_1  defaults        
                             1.2.2                    py34_1  defaults        
                             1.2.2                    py33_1  defaults        
                             1.2.2                    py33_2  defaults        
                             1.2.2                    py27_2  defaults        
                             1.2.2                    py34_2  defaults        
                             1.2.2                    py26_2  defaults        
                             1.2.3                    py34_0  defaults        
                             1.2.3                    py26_0  defaults        
                             1.2.3                    py33_0  defaults        
                             1.2.3                    py27_0  defaults        
                             1.3.1                    py27_0  defaults        
                             1.3.1                    py26_0  defaults        
                          *  1.3.1                    py33_0  defaults        
                             1.3.1                    py34_0  defaults        
                             1.3.1                    py35_0  defaults        
                             1.3.5                    py27_0  defaults        
                             1.3.5                    py35_0  defaults        
                             1.3.5                    py34_0  defaults        
                             1.4.1                    py35_0  defaults        
                             1.4.1                    py27_0  defaults        
                             1.4.1                    py34_0  defaults        
                             1.4.6                    py35_0  defaults        
                             1.4.6                    py27_0  defaults        
                             1.4.6                    py34_0  defaults        
                             1.4.8                    py35_0  defaults        
                             1.4.8                    py27_0  defaults        
                             1.4.8                    py34_0  defaults        
                             1.5                      py34_0  defaults        
                             1.5                      py35_0  defaults        
                             1.5                      py27_0  defaults        
                             1.5.1                    py36_0  defaults        
                             1.5.1                    py35_0  defaults        
                             1.5.1                    py27_0  defaults        
                             1.5.1                    py34_0  defaults        
                             1.5.4                    py36_0  defaults        
                             1.5.4                    py35_0  defaults        
                             1.5.4                    py27_0  defaults        
sphinx_rtd_theme          *  0.1.7                    py33_0  defaults        
                             0.1.7                    py26_0  defaults        
                             0.1.7                    py34_0  defaults        
                             0.1.7                    py35_0  defaults        
                             0.1.7                    py27_0  defaults        
                             0.1.9                    py34_0  defaults        
                             0.1.9                    py35_0  defaults        
                             0.1.9                    py27_0  defaults        
                             0.1.9                    py36_0  defaults        
                             0.1.9                    py36_1  defaults        
                             0.1.9                    py35_1  defaults        
                             0.1.9                    py27_1  defaults        
                             0.2.4                    py35_0  defaults        
                             0.2.4                    py27_0  defaults        
                             0.2.4                    py36_0  defaults        
```

I have also cleared the TRAVIS cache related to this PR.

**Comment by notoraptor** [comment]
jenkins test this

**Comment by notoraptor** [comment]
I am not able to reproduce Jenkins problems. I have created a branch on my computer by merging updated master branch + current PR, and it works correctly with `theano/gpuarray/tests/test_reduction.py:TestMatrix` (which failed with Jenkins: http://darjeeling.iro.umontreal.ca:8080/job/Theano_gpu/1042/testReport/junit/theano.gpuarray.tests.test_reduction/TestMatrix/test_axis_1/ ).

So, let's run Jenkins again.

@abergeron @slefrancois

**Comment by lamblin** [comment]
@abergeron has an idea for Jenkins, I'll try to fix it in this PR.

File: .jenkins/jenkins_pretest.sh [hunk 1]
```diff
@@ -8,6 +8,9 @@ set -x
 # Anaconda python
 export PATH=/usr/local/miniconda2/bin:$PATH
 
+# Clean up old .pyc files
+find ./$(git rev-parse --show-cdup) -type f -name "*.pyc" -delete
+
```
Comment (slefrancois) [inline .jenkins/jenkins_pretest.sh]: I suggest we use Jenkins' git plugin "clean before checkout" setting, which purges untracked files:

> Clean up the workspace before every checkout by deleting all untracked files and directories, including those which are specified in .gitignore. It also resets all tracked files to their versioned state. This ensures that the workspace is in the same state as if you cloned and checked out in a brand-new empty directory, and ensures that your build is not affected by the files generated by the previous build.

I have already enabled it.

Comment (lamblin) [inline .jenkins/jenkins_pretest.sh]: OK, I'll remove my commit.

**Pull request merged**